### PR TITLE
Plasmic2 sync

### PR DESCRIPTION
**Pull request by marcusellison**
I synced the project with the newly created [Plasmic studio project](https://studio.plasmic.app/projects/mTVXT6w3HHjZ4d74q3gB76)

I ran command `plasmic sync --projects mTVXT6w3HHjZ4d74q3gB76`
- _mTVXT6w3HHjZ4d74q3gB76_ is the projectId.
- this sets up the project id as the default project to sync from and pulls down any existing assets from the Plasmic studio project.

### CODE REVIEW

File: src/component/IconButton.tsx [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+//
+// interface IconButtonProps extends Omit<DefaultIconButtonProps, "hideProps1"|"hideProp2"> {
+//   // etc.
+// }
+//
+// You can also stop extending from DefaultIconButtonProps altogether and have
+// total control over the props for your component.
+interface IconButtonProps extends DefaultIconButtonProps {}
+
+function IconButton(props: IconButtonProps) {
```
Comment (xh3b4sd) [inline src/component/IconButton.tsx:24]: How does it now work when we want to structure the project our way? I would `src/component/IconButton.tsx` like to be `src/component/button/index.tsx` or something like that. I do not want to have 30 different files flying around in the same package.

**Review by xh3b4sd — COMMENTED** [review]
Some suggestions to align with our desired conventions. 

---

```
src/component/iconButton/index.tsx > src/component/iconbutton/index.tsx
```

```
src/component/plasmic/flipcard/PlasmicIconButton.css > src/component/plasmic/flipcard/index.css
```

```
src/component/plasmic/flipcard/PlasmicIconButton.tsx > src/component/plasmic/flipcard/index.tsx
```

---

`src/component/plasmic/flipcard/PlasmicMetric.css` should then actually be its own component. I want to achieve package isolation as much as this is possible. So one component per package. It looks like we throw everything together into `src/component/plasmic/flipcard` and I already do not fancy this `component/plasmic` path. When this works for you move on, though for me this is already not comprehensible anymore.

**Comment by marcusellison** [comment]
Looks like there is  a small diff in default vscode and atom prettier:
Plasmic2_sync_by_marcusellison_·_Pull_Request__7_·_venturemark_webclient

**Comment by marcusellison** [comment]
On file structure - 

We're not able to format the plasmic-maintained code in this way:
`src/component/plasmic/flipcard/PlasmicIconButton.css > src/component/plasmic/flipcard/index.css`

**Comment by xh3b4sd** [comment]
Towards the project and module structure, this is really not cool. It is nice that we can leverage code generation and get all the benefits but being restricted on how to organize ourselves is super wrong. 

---

Towards the trailing comma. I am not sure what you did or what happened. The only setting I found in my vsCode is this. We can tweak it if that is what we need. Which setting do you have for the trailing comma? 

Screenshot 2020-11-11 at 10 08 04

**Comment by marcusellison** [comment]
> Towards the project and module structure, this is really not cool. It is nice that we can leverage code generation and get all the benefits but being restricted on how to organize ourselves is super wrong.

I also want to adjust structure (we should discuss pros/cons). 
Versioning design for Venturemark can be:
 - versioning a library of components
 - versioning pages
 - versioning individual components

But one thing to keep in mind is, outside of plasmic, there is no way to version design in sync with code. Having the code of plasmic generated files portioned into 'projects' is not ideal, but it is better than the alternative - which is a complete disconnect between design and code - which easily creates redundant effort and inconsistencies.

Based on what we come to, I suggest we reach out to plasmic founders to see how they can adjust based on our preferences.

> 
> Towards the trailing comma. I am not sure what you did or what happened. The only setting I found in my vsCode is this. We can tweak it if that is what we need. Which setting do you have for the trailing comma?
> 
> Screenshot 2020-11-11 at 10 08 04

My settings were set to 'infer settings from es-lint configuration'. Disabling it solved the issue.

**Pull request merged**